extensions/arc/src/ui/dialogs/connectMiaaDialog.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import { createCredentialId } from '../../common/utils';
+import { credentialNamespace } from '../../constants';
+import * as loc from '../../localizedConstants';
+import { ControllerModel } from '../../models/controllerModel';
+import { MiaaModel } from '../../models/miaaModel';
+import { ConnectToSqlDialog } from './connectSqlDialog';
+
+export class ConnectToMiaaSqlDialog extends ConnectToSqlDialog {
+
+	constructor(private _controllerModel: ControllerModel, private _miaaModel: MiaaModel) {
+		super();
+	}
+
+	public async validate(): Promise<boolean> {
```

Comment:
You should be able to simplify this even more by putting all the connection logic here in the base class and then just have an abstract provider property that each of these implements to fill in the provider name.

`protected abstract get providerName(): string;`

then the connection profile would just be 
```
...
providerName: this.providerName,
...
```